### ISSUE TITLE
NPE with using manual Lucene indexes

### ISSUE THREAD

**Issue by antonzherdev** [orientechnologies/orientdb#7255]
## Expected behavior  
No Exception

## Actual behavior  
```
com.orientechnologies.orient.core.exception.OStorageException: Cannot put key value entry in index
	DB name="test"
com.orientechnologies.orient.core.exception.OStorageException: Cannot put key value entry in index
	DB name="test"
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doCallIndexEngine(OAbstractPaginatedStorage.java:1871)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.callIndexEngine(OAbstractPaginatedStorage.java:1838)
	at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.put(OLuceneIndexNotUnique.java:295)
	at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.put(OLuceneIndexNotUnique.java:51)
Caused by: java.lang.NullPointerException
	at com.orientechnologies.lucene.OLuceneIndexType.createField(OLuceneIndexType.java:64)
	at com.orientechnologies.lucene.builder.OLuceneDocumentBuilder.build(OLuceneDocumentBuilder.java:47)
	at com.orientechnologies.lucene.engine.OLuceneFullTextIndexEngine.buildDocument(OLuceneFullTextIndexEngine.java:232)
	at com.orientechnologies.lucene.index.OLuceneIndexNotUnique$6.callEngine(OLuceneIndexNotUnique.java:299)
	at com.orientechnologies.lucene.index.OLuceneIndexNotUnique$6.callEngine(OLuceneIndexNotUnique.java:295)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doCallIndexEngine(OAbstractPaginatedStorage.java:1860)
	... 100 more
```

## Steps to reproduce  
Create a manual Lucene index:
```
 db.getMetadata.getIndexManager.createIndex(name, itp,
          new OSimpleKeyIndexDefinition(factory.getLastVersion, OType.STRING), null, null, null,
          OLuceneIndexFactory.LUCENE_ALGORITHM)
```
Put a value there.

It's due to the commit
https://github.com/orientechnologies/orientdb/commit/8355f0b64f558527fa16a4b70c62450bc71d3800
and the line in particular:
`doc.add(createField("_CLASS", definition.getClassName(), Field.Store.YES));`

**Comment by robfrank** [comment]
I'm unable to reproduce the problem. Can you provide a more complete snippet of code?

**Comment by antonzherdev** [comment]
@robfrank I attached the code how to create the index:

```
db.getMetadata.getIndexManager.createIndex(name, itp,
          new OSimpleKeyIndexDefinition(factory.getLastVersion, OType.STRING), null, null, null,
          OLuceneIndexFactory.LUCENE_ALGORITHM)
```

Then you need to simply put a value there `idx.put("Something", doc)`

This is a manual index. My understanding is that it does not have class and definition.getClassName() returns null which is causing the NPE. There should be probably a null check.

**Comment by robfrank** [comment]
Ok, I was able to replicate the problem. I guess you are inserting in the index inside a transaction, e.g. between a begin()/commit() block. 
I fixed it, added a test case inside LuceneManualIndexTest on 2.2.x branch. Will be included in next 2.2.18. Snapshots will be available in a couple of hours
